### PR TITLE
reorganizing modules to prepare for test config schema addition - sim…

### PR DESCRIPTION
**Pull request by vsoch**
This is a first pull request to prepare for adding the test configuration schemas. I actually started on the latter, but realized in the middle that:

 1. we had an extra level of module with "tools" that made paths much longer and wasn't descriptive enough to describe anything in particular
 2. It would be simple and more intuitive to find functions based on the organization and naming of files.

The new structure looks like this:

```
$ tree -L 2 # buildtest
├── buildsystem
│   ├── base.py
│   ├── dry.py
│   ├── __init__.py
│   ├── singlesource.py
│   ├── status.py
│   └── writer.py
├── config.py
├── defaults.py
├── docgen
│   ├── __init__.py
│   └── main.py
├── __init__.py
├── log.py
├── main.py
├── menu
│   ├── build.py
│   ├── config.py
│   ├── get.py
│   ├── __init__.py
│   ├── show.py
│   └── status.py
├── settings.yml
├── system.py
└── utils
    ├── file.py
    └── __init__.py
```

And the developer navigating around the code on just the top level can directly now see the `buildsystem` module, which was hidden before, and the menu module, which is now cleanly separated.

```
├── buildsystem
├── config.py
├── defaults.py
├── docgen
├── __init__.py
├── log.py
├── main.py
├── menu
├── settings.yml
├── system.py
└── utils
```

Thus, thus PR addresses the following:

 - the 'menu' module, meaning functions to drive user interaction with the client (get, build) now has it's own module folder under `buildtest.menu`. This is intuitive because now when we move get, build, and the helper functions (e.g., writer.py) into that folder, we know they are specifically the helpers for the menu relevant to those names. Without this move, we just see these files in the general upper directory and it's not clear what they do.
 - utils has it's own folder for various utils (e.g., file.py) 
 - the defaults.py and config.py are now clearly at the top level, making it easier to find.

This structure will allow for much easier adding of the `buildtest.buildsystem.schemas` which before would have been `buildtest.tools.buildsystem.schemas` and then any sub imports of that a hugely long string! When we merge I can update the notes in the schemas repository. I'll also update any issues here with testing, since a lot of paths are changed, there are likely to be some.

Signed-off-by: vsoch <vsochat@stanford.edu>

### CODE REVIEW

**Comment by vsoch** [comment]
It's not assuring that black fails on the CI here, but I ran it locally with the same version and it seems okay. Let's ignore for now, and possibly this will be fixed when we enforce a more stable release (not one that ends with `b0`.

**Comment by vsoch** [comment]
Derp, nevermind my comment! I didn't push the black fixes, doh.

File: MANIFEST.in [hunk 1]
```diff
@@ -1,6 +1,7 @@
 include README.md LICENSE settings.yml
 recursive-include buildtest *
 graft buildtest
+include settings.yml
```
Comment (shahzebsiddiqui) [inline MANIFEST.in]: Trying to understand why we need `settings.yml` in MANIFEST.in. If i understand the first line also contains `settings.yml` and that should be removed. The `graft buildtest` in line 3 should take care of all files under `buildtest` directory. See https://docs.python.org/2/distutils/sourcedist.html#commands or https://packaging.python.org/guides/using-manifest-in/#manifest-in-commands

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]
My only one thought is I think we can move `buildtest.log` --> `buildtest.utils.log`. It's also bothering me why I have `BuildTestError` class in `log.py` while i have `BuildTestCommand` in `buildtest/tools/system.py`. I think helper classes like `BuildTestCommand`, and `BuildTestError` should be in `buildtest/utils/command.py`. Let me know if this makes sense.

File: MANIFEST.in (hunk 1 above)
Comment (vsoch) [reply MANIFEST.in]: Oh that shouldn't have been added, I was playing around with settings and forgot to remove it.
Totally agree on utils/commands.py, I can do that in another PR.

**Comment by shahzebsiddiqui** [comment]
@vsoch i am liking codecov results  👍 . I guess we did so bad percentage went from from 40% last i recall down to 22 now up to 27%.

**Comment by vsoch** [comment]
I probably won't look hard at coverage until I'm at the stage of explicitly writing tests for previously existing code. For all the new code that I'll write, I'll always pair with tests.

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]
Forgot one more on line 1 `settings.yml`

**Comment by vsoch** [comment]
The settings.yml in MANIFEST.in has been removed, per changes that were done here - the file no longer shows up in the file list. If there is some other issue with MANIFEST.in it's present in the current devel and is out of scope for this PR.

**Comment by vsoch** [comment]
Note it's in devel https://github.com/HPC-buildtest/buildtest-framework/blob/devel/MANIFEST.in#L1 and is out of scope for PR here.

**Review by shahzebsiddiqui — APPROVED** [review]
Approved!

**Comment by shahzebsiddiqui** [comment]
> Note it's in devel https://github.com/HPC-buildtest/buildtest-framework/blob/devel/MANIFEST.in#L1 and is out of scope for PR here.

I see it now.

**Pull request merged**